sapphire/core.py

```diff
@@ -59,30 +59,26 @@ def _create_listening_socket(remote, port=None, retries=10, timeout=LISTEN_TIMEO
         assert retries >= 0
         assert timeout > 0
         addr = "0.0.0.0" if remote else "127.0.0.1"
-        sock = None
+        sock = socket(AF_INET, SOCK_STREAM)
+        sock.setsockopt(SOL_SOCKET, SO_REUSEADDR, 1)
+        sock.settimeout(timeout)
         for remaining in reversed(range(retries + 1)):
```

Comment:
You could use smth like:
```
if port:
    pool = itertools.repeat(port, attempts)
else:
    pool = range(0x2770, 0xFFFF)

for port in sample(pool, k=attempts):
    ...
```

.. to be closer to ffpuppet.

Reply:
I was thinking that too but then I'd need to handle the last pass differently.